api/authentication.py

```diff
@@ -123,7 +126,13 @@ def authenticate(self, request):
         if token == "":
             raise ParseError("Missing FXA Token after 'Bearer'.")
 
-        fxa_uid = get_fxa_uid_from_oauth_token(token)
+        use_cache = True
+        method = request.method
+        if method in ["POST", "DELETE", "PUT"]:
```

Comment:
Comment (blocking): This will break the tests in #3515. Can we wait to merge #3515 and fix the tests?

Reply:
Yup, let's merge #3515 first.

Reply:
https://github.com/mozilla/fx-private-relay/blob/optimize-fxa-token-cache-mpp-3156/api/views/__init__.py#LL197C23-L197C23 
Needs to be updated to:
```
fxa_uid = get_fxa_uid_from_oauth_token(token, use_cache=False)
```
This update results in following errors:
```
FAILED api/tests/views_tests.py::TermsAcceptedUserViewTest::test_201_new_user_created_and_202_user_exists - AssertionError: Expected URL 'https://oauth.stage.mozaws.net/v1/introspect' to be called 1 times. Called 2 times.
FAILED api/tests/views_tests.py::TermsAcceptedUserViewTest::test_fxa_responds_with_no_fxa_uid_returns_404_and_cache_returns_404 - AssertionError: assert None == {'json': {'active': True, 'exp': 1686694591000}, 'status_code': 200}
FAILED api/tests/views_tests.py::TermsAcceptedUserViewTest::test_inactive_fxa_oauth_token_returns_401_and_cache_returns_401 - AssertionError: assert None == {'json': {'active': False, 'exp': 1686687391000, 'sub': 'relay-user-fxa-uid'}, 'status_code': 200}
FAILED api/tests/views_tests.py::TermsAcceptedUserViewTest::test_invalid_bearer_token_error_from_fxa_returns_500_and_cache_returns_500 - AssertionError: assert None == {'json': {'error': '401'}, 'status_code': 401}
FAILED api/tests/views_tests.py::TermsAcceptedUserViewTest::test_jsondecodeerror_returns_401_and_cache_returns_500 - AssertionError: assert None == {'json': {}, 'status_code': None}
FAILED api/tests/views_tests.py::TermsAcceptedUserViewTest::test_non_200_response_from_fxa_returns_500_and_cache_returns_500 - AssertionError: assert None == {'json': {'active': False, 'exp': 1686694593000, 'sub': 'relay-user-fxa-uid'}, 'status_code': 401}
```